<a href="https://colab.research.google.com/github/GPINTOC/Proyecto-2-Coding-Dojo/blob/main/Proyecto2_Parte1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Para esta tarea:**

Creen un cuaderno de Colab donde hayan subido y mostrado el .head() de cada uno de los conjuntos de datos.  Para cada uno de los conjuntos de datos propuestos, respondan las siguientes preguntas:

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
filename = '/content/drive/MyDrive/Documentos CodingDojo/data-ori.csv'
df = pd.read_csv(filename)
df

HAEMATOCRIT  HAEMOGLOBINS  ERYTHROCYTE  LEUCOCYTE  THROMBOCYTE   MCH  \
0            35.1          11.8         4.65        6.3          310  25.4   
1            43.5          14.8         5.39       12.7          334  27.5   
2            33.5          11.3         4.74       13.2          305  23.8   
3            39.1          13.7         4.98       10.5          366  27.5   
4            30.9           9.9         4.23       22.1          333  23.4   
...           ...           ...          ...        ...          ...   ...   
4407         32.8          10.4         3.49        8.1           72  29.8   
4408         33.7          10.8         3.67        6.7           70  29.4   
4409         33.2          11.2         3.47        7.2          235  32.3   
4410         31.5          10.4         3.15        9.1          187  33.0   
4411         33.5          10.9         3.44        5.8          275  31.7   

      MCHC    MCV  AGE SEX SOURCE  
0     33.6   75.5    1   F    out  
1     34.0   80.7    1   F    out  
2     33.7   70.7    1   F    out  
3     35.0   78.5    1   F    out  
4     32.0   73.0    1   M    out  
...    ...    ...  ...  ..    ...  
4407  31.7   94.0   92   F     in  
4408  32.0   91.8   92   F     in  
4409  33.7   95.7   93   F    out  
4410  33.0  100.0   98   F     in  
4411  32.5   97.4   99   F    out  

[4412 rows x 11 columns]

Primera opción: conjunto de datos 1

**1) Fuente de datos**
- https://www.kaggle.com/code/ashokkumarpalivela/patient-treatment-classification

**2) Breve descripción de datos**
- Data set de predicción respecto a si dejar a un paciente hospitalizado o atenderlo como ambulatorio en base a los resultados de algunas variables de hemogramas, de forma de automatizar y mejorar los tiempos de respuestas para pacientes realmente graves en la atención de salud.

**3) ¿Cuál es el objetivo?** 
- El objetivo es predecir si el paciente debe clasificarse como paciente hospitalizado o ambulatorio (Source out or in)

**4) ¿Este es un problema de clasificación o regresión?**
- Clasificación, (variables OUT or IN)

**5) ¿Cuántas características hay?**
- 11 características

**6) ¿Cuántas filas de datos hay?**
- 4412 filas 

**7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?**
- No existen datos nulos ni valores en 0, tampoco veo valores aberrantes. Lo único que me causa ruido de las variables es que hay algunas que se correlacionan directamente y no hacen sentido con el valor de la otra. El objetivo está desbalanceado (60 y 40%)

In [ ]:
filename1 = '/content/drive/MyDrive/Documentos CodingDojo/data01.csv'
df = pd.read_csv(filename1)
df

group      ID  outcome  age  gendera        BMI  hypertensive  \
0         1  125047      0.0   72        1  37.588179             0   
1         1  139812      0.0   75        2        NaN             0   
2         1  109787      0.0   83        2  26.572634             0   
3         1  130587      0.0   43        2  83.264629             0   
4         1  138290      0.0   75        2  31.824842             1   
...     ...     ...      ...  ...      ...        ...           ...   
1172      2  171130      0.0   62        1  25.516870             1   
1173      2  101659      0.0   78        1  25.822710             0   
1174      2  162069      0.0   85        2  23.891779             1   
1175      2  120967      0.0   79        2  35.288554             0   
1176      2  107636      0.0   47        1  23.121384             1   

      atrialfibrillation  CHD with no MI  diabetes  ...  Blood sodium  \
0                      0               0         1  ...    138.750000   
1                      0               0         0  ...    138.888889   
2                      0               0         0  ...    140.714286   
3                      0               0         0  ...    138.500000   
4                      0               0         0  ...    136.666667   
...                  ...             ...       ...  ...           ...   
1172                   1               0         1  ...    136.714286   
1173                   1               0         1  ...    135.680000   
1174                   1               0         1  ...    136.000000   
1175                   0               1         1  ...    140.000000   
1176                   0               0         1  ...    135.285714   

      Blood calcium    Chloride  Anion gap  Magnesium ion     PH  Bicarbonate  \
0          7.463636  109.166667  13.166667       2.618182  7.230    21.166667   
1          8.162500   98.444444  11.444444       1.887500  7.225    33.444444   
2          8.266667  105.857143  10.000000       2.157143  7.268    30.571429   
3          9.476923   92.071429  12.357143       1.942857  7.370    38.571429   
4          8.733333  104.500000  15.166667       1.650000  7.250    22.000000   
...             ...         ...        ...            ...    ...          ...   
1172      10.271429   94.428571  20.142857       2.714286    NaN    27.714286   
1173      10.523529  101.720000  18.160000       2.012500    NaN    20.480000   
1174       8.466667   97.285714  14.000000       2.028571    NaN    28.857143   
1175       8.183333  104.000000  15.750000       2.090000    NaN    24.375000   
1176       9.085714   93.857143  20.285714       2.566667    NaN    26.571429   

      Lactic acid  PCO2  EF  
0             0.5  40.0  55  
1             0.5  78.0  55  
2             0.5  71.5  35  
3             0.6  75.0  55  
4             0.6  50.0  55  
...           ...   ...  ..  
1172          NaN   NaN  40  
1173          NaN   NaN  30  
1174          NaN   NaN  55  
1175          NaN   NaN  25  
1176          NaN   NaN  55  

[1177 rows x 51 columns]

Segunda opción: conjunto de datos 2

**1) Fuente de datos**
- https://www.kaggle.com/code/laurencegesman/hospital-mortality-ml-data-analysis

**2) Breve descripción de datos**
- Data set de predicción respecto de la predicción de la mortalidad de los pacientes que ingresan a UCI a través de diferentes variables y exámenes tomados en laboratorio y signos vitales.

**3) ¿Cuál es el objetivo?** 
-  El objetivo es detectar la mortalidad hospitalaria, definida como el estado vital en el momento del alta hospitalaria en supervivientes y no supervivientes.

**4) ¿Este es un problema de clasificación o regresión?**
- Clasificación, (variables 0 or 1)

**5) ¿Cuántas características hay?**
- 51 características

**6) ¿Cuántas filas de datos hay?**
- 1177 filas 

**7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?**
- Hay datos nulos, variables categóricas de tipo 0 y 1, variables numéricas con diferentes magnitudes.